### PR TITLE
feat(require):add event listening

### PR DESCRIPTION
**Pull request by frostbane**
This patch lets the required child listen to an event fired by the parent.

This is quite long and I am writing this while on my job (secretly) so my words could be jumbled. `child`, `widget`, `feature`, `template` all mean the same thing.
#### Syntax

The syntax is quite confusing until you get used to it

``` html
<!-- parent.html -->
<rv-require name="widget"
            data-listen-☆☆="△△"></rv-require>
```

☆ is the event fired inside the widget

△ is the event fired by the parent
##### One example is:

I have a html form 

``` html
<!-- parent.html -->
<rv-require name="advancedWidget"
            data-title="show advanced settings.."
            data-listen-cleanupadvanced="reset-form"></rv-require>
<rv-require name="superAdvancedWidget"
            data-title="show super advanced settings.."
            data-listen-cleanupsuperadvanced="reset-form"></rv-require>
<input type="reset"
       value="reset everything"
       on-click="reset-button" />
```

Now when I hit the reset button I would do some clean up and and also need to tell all the required widgets to do their own clean up.

``` javascript
//parent.js

//fired when the reset button is clicked
parentRactive.on("reset-button", function(e){
    //clean up the form

    var args = { /*some argument */ };

    //broadcast it to all the required children
    parentRactive.fire("reset-form", args);
});

```

And then all the listening children could respond to it accordingly.

``` javascript
//advancedWidget.js

advancedWidgetRactive.on("cleanupadvanced", 
    function(parentEventName, eventArgs){
        //in this example the handler arguments would be
        //parenEventName = "reset-form"
        //eventArgs      = { /*some argument */}

        //do some cleaning inside the widget
    });

```

``` javascript
//superAdvancedWidget.js

superAdvancedWidgetRactive.on("cleanupsuperadvanced", 
    function(parentEventName, eventArgs){
        //in this example the handler arguments would be
        //parenEventName = "reset-form"
        //eventArgs      = { /*some argument */}

        //do some cleaning inside the widget
    });

```

The main point is I could fire a single `"reset-form"` event inside the parent and all the required children would just listen to it, and respectively do their own clean up.

Another way to call it is **broadcast** or **propagate** an event to all children.

### CODE REVIEW

**Comment by frostbane** [comment]
@XavierBoubert 

I just added this today, it could still have some bugs.

Please give me until friday to test it.

**Comment by XavierBoubert** [comment]
Sure

**Comment by frostbane** [comment]
@XavierBoubert 

There seems to be no problems. 

You can merge it anytime. d=(^o^)=b

**Comment by XavierBoubert** [comment]
Nice. I take a look this weekend and I merge it ;-)

File: features/require/require.js [hunk 1]
```diff
@@ -348,6 +355,25 @@
       });
   }
 
+  function _listenToParentEvent(parent, ractive, databinding){
+    var listenEvent = databinding.listening;
+
+    var fireLocalEvent = function(event){
+      parent.on(listenEvent[event], function(){
```
Comment (XavierBoubert) [inline features/require/require.js:362]: I think it's 

``` javascript
parent.on(event, function() {
```

Comment (XavierBoubert) [inline features/require/require.js:362]: No it's ok, my bad.

**Comment by XavierBoubert** [comment]
> The syntax is quite confusing until you get used to it

Exactly lol.

**Comment by XavierBoubert** [comment]
Merged and released in 0.6.3 ;)

**Pull request closed without merging**